### PR TITLE
[APP-6380] - support passing delete message modal

### PR DESCRIPTION
**Pull request by awlui**
Support passing delete message modal

PR description
- pass through renderRemoveMessageModal from `ChannelUI`. This allows consuming apps to supply the delete modal. The original modal will be the default.

### CODE REVIEW

File: src/modules/Channel/components/Message/index.tsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ const Message = (props: MessageProps): React.ReactElement => {
     localMessages,
   } = useChannelContext();
 
-  const { message } = props;
+  const { message, renderRemoveMessageModal = (props) => <RemoveMessageModal {...props} /> } = props;
```
Comment (karenying) [inline src/modules/Channel/components/Message/index.tsx:37]: wait why did you move this?

Comment (awlui) [reply src/modules/Channel/components/Message/index.tsx:37]: The purpose is to be able to pass the `renderRemoveMessageModal` function in from a top level parent component that we render within gather app. Prior to this change, the render function was hard coded to use the `RemoveMessageModal` component which is not customizable. This line makes it so that the component is the default but we can override by supplying our own `renderRemoveMessageModal`

**Review by mtdurling — APPROVED** [review]
LGTM - We really need to sync up our formatters with that SB was using!
